lib/controllers/share.js

```diff
@@ -103,7 +103,9 @@ module.exports = function(shareUrlPrefixes, newShareUrlPrefix) {
         makeGist(shareUrlPrefixes[newShareUrlPrefix], req.body).then(function(id) {
             id = newShareUrlPrefix + prefixSeparator + id;
             console.log('Created ID ' + id + ' using Gist service');
-            res.json({ id: id, path: req.baseUrl + '/' + id });
+            var resolveURL = req.baseUrl + '/' + id;
```

Comment:
You need to use [req.hostname](http://expressjs.com/en/api.html#req.hostname).  The host header won't be correct when there's a reverse proxy in front of our node server (which there always is in production).